### PR TITLE
Increase packer precision for bfp8 formats

### PR DESCRIPTION
**Pull request by rdjogoTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/14032)

### Problem description
bfp8 packing is inaccurate when pack_src_format is also bpf8, since this results in double rounding in the HW. First the gasket rounds to 7 bits, then rounding occurs again when the mantissas are being shifted in order to have common exponent. 

### What's changed
Add a flag to compute config called `bfp_pack_precise` which toggles the pack_src_format to either fp16 or fp32 (depending on fp32_mode_en) in order to get more accurate output. This however will half the packer bandwidth in the case of fp16, and reduce it to one quarter in the case of fp32.

### Checklist
- [x] Post commit CI passes: https://github.com/tenstorrent/tt-metal/actions/runs/11819636692
    - has same gs failure as main
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/11819637696
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/11819649013
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/11819650466
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/11819651970
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/11819653051

### CODE REVIEW

**Review by tt-aho — APPROVED** [review]
Will there be tests added to test the `bfp8_pack_precise` toggle?

**Comment by rdjogoTT** [comment]
yes, good point I will add those before merging

File: ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_program_factory.cpp [hunk 1]
```diff
@@ -119,6 +120,7 @@ UnaryProgramFactory::cached_program_t UnaryProgramFactory::create(
                 .math_fidelity = MathFidelity::HiFi4,
                 .fp32_dest_acc_en = args.fp32_dest_acc_en,
                 .unpack_to_dest_mode = unpack_to_dest_mode,
+                .bfp8_pack_precise = true,
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_program_factory.cpp:123]: It seems like this bfp_pack_precise could be inferred.

Comment (rdjogoTT) [reply ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_program_factory.cpp]: I've set it based on the op and dataformat in copy.hpp and unary.cpp

**Comment by ayerofieiev-tt** [comment]
@rdjogoTT , please update PR title

**Review by eyonland — APPROVED** [review]
It would be nice at some point to align our data type names in the code to the docs https://docs.tenstorrent.com/pybuda/latest/dataformats.html

File: tt_metal/jit_build/settings.hpp [hunk 2]
```diff
@@ -26,6 +26,7 @@ class JitBuildOptions {
     // We can keep for future WH support, otherwise not used in GS
     bool fp32_dest_acc_en;
     std::vector<UnpackToDestMode> unpack_to_dest_mode;
+    bool bfp8_pack_precise;
```
Comment (ayerofieiev-tt) [inline tt_metal/jit_build/settings.hpp:29]: To me this looks like an uninitialized member variable. 
See 
https://github.com/tenstorrent/tt-metal/blob/main/best_practices.md#15-initialize-primitive-types-on-declaration

Comment (rdjogoTT) [reply tt_metal/jit_build/settings.hpp:29]: added initial value to settings.cpp like for `fp32_dest_acc_en`

**Comment by ayerofieiev-tt** [comment]
@eyonland can someone from MCW help to bring this to binary too?

**Review by ayerofieiev-tt — APPROVED** [review]

**Pull request merged**